### PR TITLE
Update Enocean plugin

### PR DESCRIPTION
**Pull request by hasenradball**
updated and tested with the Eltako FSR61-230V.
See also Change_Log_File.

### CODE REVIEW

File: enocean/eep_parser.py [hunk 1]
```diff
@@ -13,11 +13,14 @@ def CanParse(self, eep):
         return found
 
     def Parse(self, eep, payload, status):
-        #self.logger.debug('enocean: parser called with eep={} / payload={} / status={}'.format(eep, payload, status))
+        self.logger.debug('enocean: parser called with eep = {} / payload = {} / status = {}'.format(eep, ', '.join(hex(x) for x in payload), hex(status)))
```
Comment (aschwith) [inline enocean/eep_parser.py]: This line is useful for local debugging and development. But please comment out for pull request.

File: enocean/eep_parser.py [hunk 2]
```diff
@@ -100,16 +103,16 @@ def _parse_eep_A5_04_01(self, payload, status):
...
         result['ENG'] = 0.47 + (payload[0] * 1.5 / 66)       # voltage of energy buffer in Volts
         result['HUM'] = (payload[1] / 250.0 * 100)           # relative humidity in percent
         result['TMP'] = -20.0 + (payload[2] / 250.0 * 80.0)  # temperature in degree Celsius from -20.0 degC - 60degC
         return result
 
     def _parse_eep_A5_08_01(self, payload, status):
-        # Brightness and movement sensor, for example eltako FBH65TFB, RORG = 0x07
-        #logger.debug("enocean: parsing A5_08_01: Movement sensor")
+        # Brightness and movement sensor, for example eltako FBH65TFB
+        logger.debug("enocean: parsing A5_08_01: Movement sensor")
```
Comment (aschwith) [inline enocean/eep_parser.py]: Same here, please comment out

File: enocean/eep_parser.py [hunk 3]
```diff
@@ -137,17 +140,17 @@ def _parse_eep_A5_11_04(self, payload, status):
         return results
 
     def _parse_eep_A5_12_01(self, payload, status):
-        # Status command from switche actor with powermeter, for example eltako FSVA-230, RORG = 0x07
-        #logger.debug("enocean: processing A5_12_01")
+        # Status command from switche actor with powermeter, for example eltako FSVA-230
+        logger.debug("enocean: processing A5_12_01")
```
Comment (aschwith) [inline enocean/eep_parser.py]: Comment out.

File: enocean/eep_parser.py [hunk 4]
```diff
@@ -181,21 +184,28 @@ def _parse_eep_A5_38_08(self, payload, status):
...
+
+####################################################
+### --- Definitions for RORG = D5 / ORG = 06 --- ###
+####################################################
     def _parse_eep_D5_00_01(self, payload, status):
-        #ORG = 0x06
-        #Window/Door Contact Sensor, for example Eltako FTK, FTKB
-        #logger.debug("enocean: processing D5_00_01: Door contact")
+        # Window/Door Contact Sensor, for example Eltako FTK, FTKB
+        logger.debug("enocean: processing D5_00_01: Door contact")
```
Comment (aschwith) [inline enocean/eep_parser.py]: Comment out, as mentioned before. This applies to all debug logs the you uncommented.

File: enocean/eep_parser.py [hunk 5]
```diff
@@ -243,13 +253,13 @@ def _parse_eep_F6_02_03(self, payload, status):
...
 
     def _parse_eep_F6_10_00(self, payload, status):
-        #logger.debug("enocean: processing F6_10_00: Mechanical Handle")
+        logger.debug("enocean: processing F6_10_00: Mechanical Handle")
         results = {}
         if (payload[0] == 0xF0):
             results['STATUS'] = 0
         elif ((payload[0]) == 0xE0) or ((payload[0]) == 0xC0):
             results['STATUS'] = 1
-        # Typo error in older Eltako Datasheet for 0x0D instead of the right 0xD0
```
Comment (aschwith) [inline enocean/eep_parser.py]: Please revert your changes. There was in fact a typo (Schreibfehler) not a type errror.

File: enocean/__init__.py [hunk 6]
```diff
@@ -628,49 +711,118 @@ def send_rgbw_dim(self,id_offset=0, color='red', dim=0, dimspeed=0):
...
+            return None
+        self.logger.info('enocean: sending learn telegram for rgbw dim command with [ID-Offset], [RORG], [payload] / [{:#04x}], [{:#04x}], [{}]'.format(id_offset, rorg, ', '.join('{:#04x}'.format(x) for x in payload)))
+        self._send_radio_packet(id_offset, rorg, payload)
+
 
     def send_learn_switch(self, id_offset=0):
+        self.logger.debug('enocean: send_learn_switch()')
+        # Prepare Data
+        rorg = 0xA5
+        payload = [0xE0, 0x40, 0x0D, 0x80]
```
Comment (aschwith) [inline enocean/__init__.py]: Please explain the requested payload changes. Which devices did you test? The current [0x01, 0x00, 0x00, 0x00] works fine with all Eltako switches. Perhaps we have to introduce a new learn switch command?

File: enocean/Enocean_Change_Log.txt [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+#######
+
+# __init__.py
+
+	1.) Change learn command for send_learn_switch:
+		0xA5, [0xE0, 0x40, 0x0D, 0x80]
+	
+	2.) Change bugs in send_learn_switch method
+	
+	3.) change org (0x07) to RORG (0xA5) in send_learn_dim method
```
Comment (aschwith) [inline enocean/Enocean_Change_Log.txt]: Cannot find the change from org 0x07 to RORG 0xA5 in the code. Please doublecheck.

File: enocean/__init__.py (hunk 6 above)
Comment (aschwith) [inline enocean/__init__.py]: Looks right. Source: https://www.eltako.com/fileadmin/downloads/de/Gesamtkatalog/Eltako_Gesamtkatalog_KapT_low_res.pdf. Approved, closing this one.

**Comment by hasenradball** [comment]
close

**Pull request closed without merging**